c2rust-refactor/src/context.rs

```diff
@@ -1317,15 +1321,21 @@ impl<'a, 'tcx, 'b> TypeCompare<'a, 'tcx, 'b> {
                     return true;
                 }
 
-                def1.all_fields().count() == def2.all_fields().count()
+                def1.variants().len() == def2.variants().len()
```

Comment:
`.all_fields()` does a flat map over `.variants()` and then `.fields()`, which is what we're doing here. Is the issue that the flattened iterator may have the same elements, but the boundaries of each variant are different?

Reply:
That's right, plus that it doesn't account correctly for empty variants. For example, it considers the following two enums equivalent:
```rust
enum Foo {
  One,
}
enum Bar {
  One,
  Two,
}
```